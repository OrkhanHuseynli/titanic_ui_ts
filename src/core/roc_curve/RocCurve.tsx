import * as React from 'react';
// @ts-ignore
import rd3 from 'react-d3-library'


export interface RocCurveProps {
    points : number[][];
}


export const buildRocCurve: React.FunctionComponent<RocCurveProps> = (props) => {

    return <h1>{props.points}</h1>;
};