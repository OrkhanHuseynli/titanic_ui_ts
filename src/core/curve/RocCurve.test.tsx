import {buildRocCurve, RocCurveProps} from "./RocCurve";
import {configure, shallow, ShallowWrapper} from "enzyme";
import { create, ReactTestRenderer } from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16'

import * as React from "react";

let wrapper: ShallowWrapper<RocCurveProps>;
let snapshot: ReactTestRenderer;

beforeEach(() => {
    const mockPoints = {points: [[3,4],[4,5]]}
    const mockRocCurve = <h1>{mockPoints.points}</h1>;

    wrapper = shallow(mockRocCurve);
    snapshot = create(mockRocCurve);
});

configure({adapter: new Adapter()});

describe('buildRocCurve', () => {
    test('it matches the snapshot', () => {
        expect(snapshot.toJSON()).toMatchSnapshot();
    });

    // it('it should toggle checkbox label after click event', () => {
    //     expect(wrapper.text()).toEqual('Off');
    //
    //     wrapper.find('input').simulate('change');
    //
    //     expect(wrapper.text()).toEqual('On');
    // });
});