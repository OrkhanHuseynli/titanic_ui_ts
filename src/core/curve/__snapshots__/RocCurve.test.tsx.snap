// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buildRocCurve it matches the snapshot 1`] = `
<h1>
  3
  4
  4
  5
</h1>
`;
